Index: SocialMedia.Application/DTOs/CreatePostDto.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>namespace SocialMedia.Application.DTOs;\r\n\r\npublic record CreatePostDto(string Text);
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SocialMedia.Application/DTOs/CreatePostDto.cs b/SocialMedia.Application/DTOs/CreatePostDto.cs
--- a/SocialMedia.Application/DTOs/CreatePostDto.cs	(revision b5182819f100bb7e3ea537c14fd02796e0324f5a)
+++ b/SocialMedia.Application/DTOs/CreatePostDto.cs	(date 1744306772417)
@@ -1,3 +1,5 @@
+using System.ComponentModel.DataAnnotations;
+
 namespace SocialMedia.Application.DTOs;
 
-public record CreatePostDto(string Text);
\ No newline at end of file
+public record CreatePostDto([Required] string Text);
\ No newline at end of file
Index: SocialMedia.Presentation/Controllers/PostsController.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/SocialMedia.Presentation/Controllers/PostsController.cs b/SocialMedia.Presentation/Controllers/PostsController.cs
new file mode 100644
--- /dev/null	(date 1744307672148)
+++ b/SocialMedia.Presentation/Controllers/PostsController.cs	(date 1744307672148)
@@ -0,0 +1,54 @@
+using Infrastructure;
+using Microsoft.AspNetCore.Mvc;
+using SocialMedia.Application.Contracts;
+
+namespace SocialMedia.Controllers;
+
+[Route("api/posts")]
+[ApiController]
+public class PostsController : ControllerBase
+{
+    private readonly IPostService _postService;
+    
+    public PostsController(IPostService postService)
+    {
+        _postService = postService;
+    }
+    
+
+    [HttpGet]
+    public async Task<IActionResult> GetPost([FromRoute] int postId)
+    {
+        var result = await _postService.GetPost(postId);
+
+        return result switch
+        {
+            
+        };
+    }
+
+    [HttpGet]
+    public IActionResult<IActionResult> GetPostsOfUsername([FromQuery] string username)
+    {
+        
+    }
+
+    public IActionResult<IActionResult> UpdatePost()
+    {
+        
+    }
+    
+    public IActionResult<IActionResult> CreatePost()
+    {
+        
+    }
+    
+    public IActionResult<IActionResult> DeletePost()
+    {
+        
+    }
+    // Get Posts by Active Status
+    // Update Post
+    // Create Post
+    // Delete Post
+}
\ No newline at end of file
